bpf/lib/nat.h

```diff
@@ -421,6 +415,9 @@ snat_v4_icmp_rewrite_ingress_embedded(struct __ctx_buff *ctx,
 		return DROP_WRITE_ERROR;
 	if (ipv4_csum_update_by_diff(ctx, l4_off + sizeof(struct icmphdr), sum) < 0)
 		return DROP_CSUM_L3;
+	if (csum.offset &&
+	    csum_l4_replace(ctx, inner_l4_off, &csum, 0, sum, flags) < 0)
```

Comment:
In the case of UDP, this `csum_l4_replace` will run for twice. The first run is inside the `l4_modify_port` as you revised, the second run is here as `csum.offset` is non-zero.